### ISSUE TITLE
False-positive excessive_precision on 1_000_000_000.0

### ISSUE THREAD

**Issue by frol** [rust-lang-nursery/rust-clippy#2715]
The following code triggers `excessive_precision`:

```rust
fn main() {
    let y = 10f32;
    let x = y * 1_000_000_000.0;
    println!("{}", x);
}
```

```
warning: float has excessive precision
 --> src/main.rs:3:17
  |
3 |     let x = y * 1_000_000_000.0;
  |                 ^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `1000000000`
  |
```

**Comment by estk** [comment]
@frol thanks for calling this one out.
This is actually an interesting case since the `.0` is really acting as a stand in for `f32` and still of course it is technically "excessive precision". The `.0` is common enough that i would be inclined to just ignore the error for this special case.
The other possibility is changing the help message to indicate that the programmer should use the postfix of f32 to indicate that the literal in question is in fact a float.
What do you think @oli-obk ?

**Comment by oli-obk** [comment]
Yea we should not lint this case

**Comment by estk** [comment]
I can take this issue btw

**Comment by DmitriK** [comment]
Should the help message still be updated further for this issue? Got this lint when trying out `clippy-preview` today, and was confused as the help message proposes removing the `.` completely, but that just causes a compile error in my case and in the original example. 

Additionally, I'm used to adding a single `.` (e.g. `1_000_000_000.`) for float literals, which the lint still suggests to remove. I understand if a single `.` would be considered non-idiomatic, but I feel like the help should suggest changing it to `.0`.

**Comment by flip1995** [comment]
Thanks for pointing this out! I opened a new issue for this.
